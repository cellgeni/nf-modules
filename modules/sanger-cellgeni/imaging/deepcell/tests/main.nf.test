// nf-core modules test imaging/deepcell
nextflow_process {

    name "Test Process IMAGING_DEEPCELL"
    script "../main.nf"
    process "IMAGING_DEEPCELL"

    tag "modules"
    tag "modules_sanger"
    tag "imaging"
    tag "imaging/deepcell"

    test("mouse_heart - tiff") {

        when {
            process {
                """
                input[0] = [
                    [ id:'mouse_heart'],
                    0, 0, 600, 600,
                    file(params.modules_testdata_base_path + 'imaging/tiff/mindagap.mouse_heart.wga.tiff', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("mouse_heart - tiff - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'mouse_heart'],
                    0, 0, 200, 200,
                    file(params.modules_testdata_base_path + 'imaging/tiff/mindagap.mouse_heart.wga.tiff', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
