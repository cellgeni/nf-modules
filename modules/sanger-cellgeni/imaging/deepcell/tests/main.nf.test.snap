{
    "mouse_heart - tiff - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "mouse_heart"
                        },
                        "mouse_heart_0_0_200_200_deepcell_outlines.wkt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,600c2d8fa5e77e818bb6e66726ba2f96"
                ],
                "versions": [
                    "versions.yml:md5,600c2d8fa5e77e818bb6e66726ba2f96"
                ],
                "wkts": [
                    [
                        {
                            "id": "mouse_heart"
                        },
                        "mouse_heart_0_0_200_200_deepcell_outlines.wkt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-02-11T09:33:39.654173115"
    },
    "mouse_heart - tiff": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "mouse_heart"
                        },
                        "mouse_heart_0_0_600_600_deepcell_outlines.wkt:md5,53d08480bbef4b8349ba70ebef3fc5dc"
                    ]
                ],
                "1": [
                    "versions.yml:md5,600c2d8fa5e77e818bb6e66726ba2f96"
                ],
                "versions": [
                    "versions.yml:md5,600c2d8fa5e77e818bb6e66726ba2f96"
                ],
                "wkts": [
                    [
                        {
                            "id": "mouse_heart"
                        },
                        "mouse_heart_0_0_600_600_deepcell_outlines.wkt:md5,53d08480bbef4b8349ba70ebef3fc5dc"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-02-11T09:33:16.099316588"
    }
}